Makefile.toml

```diff
@@ -10,8 +10,28 @@ BUILDSYS_TOOLS_DIR = "${BUILDSYS_ROOT_DIR}/tools"
 BUILDSYS_SOURCES_DIR = "${BUILDSYS_ROOT_DIR}/sources"
 BUILDSYS_TIMESTAMP = { script = ["date +%s"] }
 BUILDSYS_VERSION_BUILD = { script = ["git describe --always --dirty --exclude '*' || echo 00000000"] }
-BUILDSYS_VERSION_IMAGE = { script = ["awk -F '[ =\"]+' '$1 == \"version\" {print $2}' Release.toml"] }
+# For now, release config path can't be overridden with -e, because it's used
+# later in this section.  You have to edit the path here in Makefile.toml to
+# use a different Release.toml.
+BUILDSYS_RELEASE_CONFIG_PATH = "${BUILDSYS_ROOT_DIR}/Release.toml"
+BUILDSYS_VERSION_IMAGE = { script = ["awk -F '[ =\"]+' '$1 == \"version\" {print $2}' ${BUILDSYS_RELEASE_CONFIG_PATH}"] }
+# This can be overridden with -e to build a different variant from the variants/ directory
 BUILDSYS_VARIANT = "aws-k8s-1.15"
+# Product name used for file and directory naming
+BUILDSYS_NAME = "bottlerocket"
+
+# These can be overridden with -e to change configuration for pubsys (`cargo
+# make repo`).  In addition, you can set RELEASE_START_TIME to determine when
+# update waves and repo metadata expiration times will start, instead of
+# starting now.  (This can be an RFC3339 date, or an offset like "in X
+# hours/days/weeks".)
+REPO_EXPIRATION_POLICY_PATH = "${BUILDSYS_ROOT_DIR}/tools/pubsys/policies/repo-expiration/2w-2w-1w.toml"
+REPO_WAVE_POLICY_PATH = "${BUILDSYS_ROOT_DIR}/sources/updater/waves/default-waves.toml"
+REPO_INFRA_CONFIG_PATH = "${BUILDSYS_ROOT_DIR}/Infra.toml"
+# Default repo to read from REPO_INFRA_CONFIG_PATH
+REPO = "default"
+# Default signing key to use from REPO_INFRA_CONFIG_PATH
+KEY = "default"
```

Comment:
I'd rather go the other direction and remove the prefix from "BUILDSYS_*" for variables that the user is likely to change, like variant.  That was our methodology here - short variables for things likely to be overridden.  I think it kind of sucks to have to remember these prefixes, especially when you otherwise have no idea what buildsys or pubsys are, since you just deal with "cargo make".

These don't get set by environment variables, only `-e`, so I'm not as concerned about namespacing.  I just tested exporting a bogus BUILDSYS_VARIANT in my shell and it's ignored.

Reply:
The prefixed form is the current convention in the makefile, so I'd like to stick to it.

The prefixes will disambiguate it from other future use cases, so we don't end up with `KEY` (for TUF repos) and `BUILDSYS_KEY` (for signed boot) or similar. They're also easier to grep for when refactoring, and they don't match unpacked source archives that might be present in the tree.

Reply:
Then I propose that we rename them `BUILD_` and `REPO_` because, again, no one knows (or should have to know) what buildsys and pubsys are.  We could start with `REPO_` in this PR and fix buildsys soon after.

Reply:
`BUILD_` and `PUBLISH_` perhaps? `REPO_REPO` would be odd.

Reply:
Well, we'd go with something not-odd like `REPO_NAME` instead, but I think `PUBLISH_` is fine because `REPO_` implies the existence of `AMI_`/`SSM_`, but we'd want more generic names for those to cover other platforms, and I can't think of other names atm.